c_src/sqlite3_nif.c

```diff
@@ -785,6 +861,18 @@ on_load(ErlNifEnv* env, void** priv, ERL_NIF_TERM info)
 {
     assert(env);
 
+    static const sqlite3_mem_methods methods = {
+      exqlite_malloc,
+      exqlite_free,
+      exqlite_realloc,
+      exqlite_mem_size,
+      exqlite_mem_round_up,
+      exqlite_mem_init,
+      exqlite_mem_shutdown,
+      0};
+
+    sqlite3_config(SQLITE_CONFIG_MALLOC, &methods);
```

Comment:
Before we do this I suggest adding a global variable in the top of the file:
```
static sqlite3_mem_methods default_alloc_methods;
```

then
```suggestion
    sqlite3_config(SQLITE_CONFIG_GETMALLOC, &default_alloc_methods);
    sqlite3_config(SQLITE_CONFIG_MALLOC, &methods);
```

Reply:
Implemented! Thanks